apps/laboratory/src/components/RandomLink.tsx

```diff
@@ -1,12 +1,17 @@
 'use client'
 
 import Link from 'next/link'
+import { useState, useEffect } from 'react'
 
 export function RandomLink({ hrefs, children }: { hrefs: string[]; children: React.ReactNode }) {
-  const href = hrefs[Math.floor(Math.random() * hrefs.length)]
-  if (!href) {
-    throw new Error('No hrefs provided')
-  }
+  const [href, setHref] = useState<string>()
+  useEffect(() => {
+    const newHref = hrefs[Math.floor(Math.random() * hrefs.length)]
+    if (!href) {
+      throw new Error('No hrefs provided')
+    }
```

Comment:
@chris13524  @tomiir this is throwing the error as state of href is always `undefined`. State of href is not getting updated as we are throwing error. Tried running the lab locally an it failed here.

Reply:
https://github.com/WalletConnect/web3modal/pull/2492